http-clients/src/main/java/org/triplea/http/client/web/socket/GenericWebSocketClient.java

```diff
@@ -43,9 +43,9 @@ public GenericWebSocketClient(final URI lobbyUri, final Consumer<String> errorHa
   }
 
   @VisibleForTesting
-  static URI swapHttpsToWssProtocol(final URI uri) {
-    return uri.getScheme().equals("https")
-        ? URI.create(uri.toString().replace("https", "wss"))
+  static URI swapHttpToWsProtocol(final URI uri) {
+    return uri.getScheme().matches("^https?$")
+        ? URI.create(uri.toString().replaceFirst("^http", "ws"))
```

Comment:
Not sure if you're getting a bit over-paranoid with `replaceFirst`, the regex `^` really guarantees one match here. 

It seems like we are defending against the protocol from being `shttp`? Otherwise the protocol string being equal to https was surefire already.

Do we have to replace `http` protocol with `ws`?

Reply:
Yes, because the JDK API requires the protocol to be wss, http doesn't work